docs: document how to upload a file for a project using the API (#1623)
keptn/keptn.github.io: content/docs/1.0.x/reference/api/index.md

    CLI is authenticated against the Keptn cluster http://YOUR.DOMAIN/api
    ```

* Access the Keptn Swagger UI and API documentation

> * by clicking on "Keptn API" in the top right menu on the Bridge
> * or by pointing your browser at: http://YOUR.DOMAIN/api .

Reviewer: This way, it is indented and not at the same level and the last paragraph is not belonging to a bullet point as on the screenshot :) 
![image](https://github.com/keptn/keptn.github.io/assets/992621/c176967b-5cca-4ba9-973e-3f50131ca500)

```suggestion
    * by clicking on "Keptn API" in the top right menu on the Bridge
    * or by pointing your browser at: http://YOUR.DOMAIN/api .

```

    CLI is authenticated against the Keptn cluster http://YOUR.DOMAIN/api
    ```

> * Access the Keptn Swagger UI and API documentation from the Bridge
> * by clicking on "Keptn API" in the top right menu
> * or by pointing your browser at: http://YOUR.DOMAIN/api .

Reviewer 2: ```suggestion
* Access the Keptn Swagger UI and API documentation
* by clicking on "Keptn API" in the top right menu on the Bridge
* or by pointing your browser at: http://YOUR.DOMAIN/api .
This gives you easy access to all API Endpoints.
```

        link="./assets/swagger-example.png"
        caption="Keptn Swagger API documentation - Example"
        width="700px">}}
        
## Uploading an API file to a Keptn project 
* An API file can be uploaded to a project in either of the following ways :

>     * Upload the file directly to your git repo.

Reviewer 2: ```suggestion
    * Upload the file directly to your Git repo.
```

>     * Use the `resource-service` API for uploading, as shown in the example snippet from above.

Reviewer 2: ```suggestion
* Use the `resource-service` API for uploading
using the Swagger API as shown above.
This gives you sections for uploading resources
on the project, stage or service level.

  To mimic what the `keptn add-resource` command does,
  use the `POST API` call.
```
You might want to grab the screenshot that Andi provided and insert it here but I'm not going to try to do that in the github editor ;-)

        link="./assets/swagger-example.png"
        caption="Keptn Swagger API documentation - Example"
        width="700px">}}
        
## Upload a file to a Keptn project
You can upload an API file to a project in any of the following ways :

> * Use the [keptn add-resource](../cli/commands/keptn_add-resource) command.

Reviewer 2: ```suggestion
* Use the [keptn add-resource](../cli/commands/keptn_add-resource/) command.
```

> * Use the [keptn add-resource](../../cli/commands/keptn_add-resource) command.

Reviewer 2: ```suggestion
* Use the [keptn add-resource](../cli/commands/keptn_add-resource) command.
```

> * Use the [keptn add-resource](../../../cli/commands/keptn_add-resource) command.

Reviewer 2: ```suggestion
* Use the [keptn add-resource](../../../cli/commands/keptn_add-resource/) command.
```

Note that only some APIs are supported by this command.
* Upload the file directly to your Git repo.
* Use the `resource-service` API for uploading
using the Swagger API as shown above.
This gives you sections for uploading resources
on the project, stage or service level.

  To mimic what the `keptn add-resource` command does,
  use the `POST API` call.
  
  ```console
  curl -X 'POST' \

>   'http://yourkeptn/api/resource-service/v1/project/YOURPROJECT/stage/YOURSTAGE/service/YOURSERVICE/resource' \

Reviewer: For consistency with the rest of "parameters"
```suggestion
  'http://YOURKEPTN/api/resource-service/v1/project/YOURPROJECT/stage/YOURSTAGE/service/YOURSERVICE/resource' \
```

        link="./assets/swagger-example.png"
        caption="Keptn Swagger API documentation - Example"
        width="700px">}}
        

> ## Uploading an API file to a Keptn project 

Reviewer 3: The term "API file" doesn't make sense here. It should just be "upload a file to a Keptn project".

You can upload an API file to a project in any of the following ways :
* Use the [keptn add-resource](../../cli/commands/keptn_add-resource) command.
Note that only some APIs are supported by this command.
* Upload the file directly to your Git repo.
* Use the `resource-service` API for uploading
using the Swagger API as shown above.
This gives you sections for uploading resources
on the project, stage or service level.

  To mimic what the `keptn add-resource` command does,
  use the `POST API` call.
## Architecture Details of Keptn API

> Keptn 0.6.2 introduced an NGINX as new K8s deployment and service called `api-gateway-ngnix` shown in the figure below. This NGINX allows to route the traffic and ensures that all requests are authenticated using the `/auth` endpoint of the `api-service`.

Reviewer 3: Keptn 0.6.2 was a **very** long time ago (almost 3 years ago). I don't think this is required information anymore.

Author: Okay, @agardnerIT. I will remove that.

Reviewer 2: Not so fast!  The ancient history references aren't needed but what it describes is still relevant!  I'll suggest a rewrite.  And note that this comment is not related to anything @Bhavya wrote -- just a sign that all this doc is overdue for a review ;-)

        link="./assets/swagger-example.png"
        caption="Keptn Swagger API documentation - Example"
        width="700px">}}
        
## Upload a file to a Keptn project
You can upload an API file to a project in any of the following ways :
* Use the [keptn add-resource](../../cli/commands/keptn_add-resource) command.
Note that only some APIs are supported by this command.
* Upload the file directly to your Git repo.
* Use the `resource-service` API for uploading
using the Swagger API as shown above.
This gives you sections for uploading resources
on the project, stage or service level.

  To mimic what the `keptn add-resource` command does,
  use the `POST API` call.
  
  ```console
  curl -X 'POST' \
  'http://yourkeptn/api/resource-service/v1/project/YOURPROJECT/stage/YOURSTAGE/service/YOURSERVICE/resource' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -d '{
  "resources": [
    {
      "resourceContent": "PATHTO/yourfile.abc",
      "resourceURI": "BASE64ENCODEDFILE"
    }
  ]
  }
  ```
## Architecture Details of Keptn API

> Keptn 0.6.2 introduced an NGINX as new K8s deployment and service called `api-gateway-ngnix` shown in the figure below. This NGINX allows to route the traffic and ensures that all requests are authenticated using the `/auth` endpoint of the `api-service`.

Reviewer 2: ```suggestion
Keptn uses
[NGINX](https://www.nginx.com/)
to route the traffic and ensure that all requests are authenticated
using the `/auth` endpoint of the `api-service`.
This is implemented using the `api-gateway-ngnix` service
as shown in the figure below.
```
The history is no longer relevant but the implementation is of interest.  I also broke this up into shorter lines in case someone turns on markdownlint for this repo.

Reviewer 2: The bullet list that followed this text also needed updating -- it talks about the configuration-service, which was replaced by the resource-service in 0.16.x.  I couldn't get to that text to make "suggestions" so just did an edit and committed it to your branch.  While I was at it, I linked to the main Architecture document for some stuff and I hope I got the right number of ../ strings in -- I can't check xrefs when using the github editor.  I guess the xref checker will flag them if I got them wrong.

Author: okay, Except for this, I guess everything is okay! What's your point of view? @StackScribe
